### PR TITLE
Embed torch.nn typing annotations

### PR DESCRIPTION
**Pull request by malfet**
Delete several .pyi files and embed annotations from those files in respective .py

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/nn/utils/spectral_norm.py [hunk 1]
```diff
@@ -3,20 +3,26 @@
...
-    _version = 1
+    _version: int = 1
     # At version 1:
     #   made  `W` not a buffer,
     #   added `v` as a buffer, and
     #   made eval mode use `W = u @ W_orig @ v` rather than the stored `W`.
+    name: str
+    dim: int
+    n_power_iterations: int
+    eps: float
```
Comment (rgommers) [inline torch/nn/utils/spectral_norm.py:23]: These seem unnecessary, `mypy` is happy with only the annotations of `__init__()` as far as I can tell. Adding these is harmless, but we want to make a habit of this?

**Review by rgommers — APPROVED** [review]
LGTM, just one question about class attributes.

File: torch/nn/utils/spectral_norm.py (hunk 1 above)
Comment (malfet) [reply torch/nn/utils/spectral_norm.py:23]: This PR just copies existing annotations.
I thought this is done to annotate class members, which are likely to be accessed outside of the class scope.

Comment (rgommers) [reply torch/nn/utils/spectral_norm.py:23]: It's covered in PEP 526; several methods are valid it looks like. One could also leave these out, or use
```
self.name: str = name
```
inline instead. I don't have a strong preference either way.

**Pull request closed without merging**